cogeo_mosaic/backends/base.py

```diff
@@ -0,0 +1,61 @@
+"""cogeo_mosaic.backend.base: base Backend class."""
+
+import abc
+from contextlib import AbstractContextManager
+from typing import Dict, Optional, Tuple
+
+from cogeo_mosaic import version as mosaic_version
+from cogeo_mosaic.backends.utils import get_hash
+from cogeo_mosaic.model import MosaicJSON
+
+
+class BaseBackend(AbstractContextManager):
+    mosaic_def: MosaicJSON
+
+    @abc.abstractmethod
+    def __init__(self, *args, **kwargs):
+        """Connect to backend"""
+
+    def __enter__(self):
+        """Support using with Context Managers"""
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        """Support using with Context Managers"""
+
+    def metadata(self) -> Dict:
+        """Retrieve Mosaic metadata
+
+        Returns
+        -------
+        MosaicJSON as dict without `tiles` key.
+        """
+        return {k: v for k, v in dict(self.mosaic_def).items() if k != "tiles"}
+
+    @abc.abstractmethod
+    def tile(self, x: int, y: int, z: int) -> Tuple[str]:
+        """Retrieve assets for tile."""
+
+    @abc.abstractmethod
+    def point(self, lng: float, lat: float) -> Tuple[str]:
+        """Retrieve assets for point."""
+
+    @abc.abstractmethod
+    def read_mosaic(self, *args, **kwargs) -> MosaicJSON:
+        """Fetch mosaic definition"""
+
+    @property
+    def mosaicid(self) -> str:
+        return get_hash(body=dict(self.mosaic_def), version=mosaic_version)
```

Comment:
I don't think we need to add the `version` here, the mosaicid should be uniq per mosaic, by adding the version, each time the cogeo-mosaic module will be updated the mosaic id will change.

Reply:
Yes I agree. I only add the version because you did so before. In fact the hash computed here still won't match one computed from the command line, because it's wrapped in another dict `{'body': ...`

Reply:
so just do 
```
get_hash(**dict(self.mosaic_def)) 
```
?

Reply:
Yes, that should be equivalent.